### PR TITLE
JAMES-3452 Implement Identity/get

### PR DESCRIPTION
**Pull request by chibenwa**
Based on top of CanSendFrom API it allows a JMAP client
to discover which addresses it can use and would need to
position on EmailSubmissions...

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

File: server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/rfc8621/RFC8621MethodsModule.java [hunk 1]
```diff
@@ -32,19 +32,7 @@
...
-import org.apache.james.jmap.method.EmailSubmissionSetMethod;
-import org.apache.james.jmap.method.MailboxGetMethod;
-import org.apache.james.jmap.method.MailboxQueryMethod;
-import org.apache.james.jmap.method.MailboxSetMethod;
-import org.apache.james.jmap.method.Method;
-import org.apache.james.jmap.method.SystemZoneIdProvider;
-import org.apache.james.jmap.method.VacationResponseGetMethod;
-import org.apache.james.jmap.method.VacationResponseSetMethod;
-import org.apache.james.jmap.method.ZoneIdProvider;
+import org.apache.james.jmap.method.*;
```
Comment (Arsnael) [inline server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/rfc8621/RFC8621MethodsModule.java]: wildcard spotted

**Review by Arsnael — APPROVED** [review]

File: server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/rfc8621/RFC8621MethodsModule.java (hunk 1 above)
Comment (chibenwa) [reply server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/rfc8621/RFC8621MethodsModule.java]: outdated :-)

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSubmissionSet.scala [hunk 2]
```diff
@@ -87,4 +87,5 @@ object EmailSubmissionCreationRequest {
 }
 
 case class EmailSubmissionCreationRequest(emailId: MessageId,
+                                          identityId: Option[Id],
```
Comment (rouazana) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSubmissionSet.scala:90]: no associated test?

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSubmissionSet.scala:90]: That's contributed on my personal time. I did not took the time to do it.

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSubmissionSet.scala:90]: Very good catch BTW

**Pull request closed without merging**